### PR TITLE
Add latex block

### PR DESCRIPTION
**Pull request by d471061c**
fixes: #94

Here's a draft of what the editor would look like:
latest_latex

### CODE REVIEW

File: services/cms/src/blocks/Latex/LatexEditor.tsx [hunk 1]
```diff
@@ -0,0 +1,59 @@
+import { css } from "@emotion/css"
+import styled from "@emotion/styled"
+import { BlockEditProps } from "@wordpress/blocks"
+import KaTex from "katex"
+import React from "react"
+
```
Comment (nygrenh) [inline services/cms/src/blocks/Latex/LatexEditor.tsx:6]: import katex css here

**Review by nygrenh — CHANGES_REQUESTED** [review]
+ block renderer implementation in course material

**Review by nygrenh — CHANGES_REQUESTED** [review]
```js
import "katex.css"
const LatexP = styled.p``

export default LatexP
```

**Review by nygrenh — CHANGES_REQUESTED** [review]
dynamic https://nextjs.org/docs/advanced-features/dynamic-import

File: system-tests/src/tests/latex.spec.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  // Click text=Add course
+  await page.click("text=Add course")
+  // Click input[type="text"]
+  await page.click('input[type="text"]')
+  // Fill input[type="text"]
+  await page.fill('input[type="text"]', "Latex course")
+  // Click text=Create course
+  await page.click("text=Create course")
+  // Click :nth-match(:text("Manage"), 2)
+  await page.click(':nth-match(:text("Manage"), 2)')
```
Comment (nygrenh) [inline system-tests/src/tests/latex.spec.ts:24]:
```js
await Promise.all([
    page.waitForNavigation(),
    page.click(':nth-match(:text("Manage"), 2)')
  ])
```

File: system-tests/src/tests/latex.spec.ts [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+  expectPath(page, "/manage/courses/[id]")
+  // Click text=Manage pages
+  await page.click("text=Manage pages")
```
Comment (nygrenh) [inline system-tests/src/tests/latex.spec.ts:28]: Promise.all

File: system-tests/src/tests/latex.spec.ts [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+  // Click [placeholder="Search"]
+  await page.click('[placeholder="Search"]')
+  // Fill [placeholder="Search"]
+  await page.fill('[placeholder="Search"]', "latex")
+  // Click button[role="option"]:has-text("Latex Block")
+  await page.click('button[role="option"]:has-text("Latex Block")')
+  // Fill textarea
+  await page.fill("textarea", "\\mathbb{R}")
+  // Click button:has-text("Save")
+  await page.click('button:has-text("Save")')
```
Comment (nygrenh) [inline system-tests/src/tests/latex.spec.ts:56]: wait for saved notification

File: system-tests/src/tests/latex.spec.ts [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+  // Click text=Home
+  await Promise.all([
+    page.waitForNavigation(/*{ url: 'http://project-331.local/' }*/),
+    page.click("text=Home"),
+  ])
+  // Click text=University of Helsinki, Department of Mathematics and Statistics
+  await page.click("text=University of Helsinki, Department of Mathematics and Statistics")
```
Comment (nygrenh) [inline system-tests/src/tests/latex.spec.ts:63]: promise.all

File: system-tests/src/tests/latex.spec.ts [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+  expectPath(page, "/organizations/[id]")
+  // Click text=Latex course
+  await Promise.all([
+    page.waitForNavigation(/*{ url: 'http://project-331.local/courses/latex-course' }*/),
+    page.click("text=Latex course"),
+  ])
+  // Click button:has-text("Continue")
+  await page.click('button:has-text("Continue")')
+  // Click text=Chapter 1: first page
+  await page.click("text=Chapter 1: first page")
```
Comment (nygrenh) [inline system-tests/src/tests/latex.spec.ts:73]: promise.all

File: system-tests/src/tests/latex.spec.ts [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+  expectPath(page, "/courses/latex-course/chapter-1")
+  // Click :nth-match(span:has-text("R"), 3)
+  await page.waitForSelector(':nth-match(span:has-text("R"), 3)')
+})
```
Comment (nygrenh) [inline system-tests/src/tests/latex.spec.ts:77]: screenshot?

File: system-tests/src/tests/latex.spec.ts [hunk 8]
```diff
@@ -0,0 +1,77 @@
+import { expect, test } from "@playwright/test"
+
+import expectPath from "../utils/expect"
+
+test.use({
+  storageState: "src/states/admin@example.com.json",
+})
+
+test("latex-block renders", async ({ page }) => {
```
Comment (nygrenh) [inline system-tests/src/tests/latex.spec.ts:9]: Also write a paragraph that contains latex and screenshot that as well

**Review by nygrenh — CHANGES_REQUESTED** [review]

File: system-tests/src/tests/latex.spec.ts [hunk 9]
```diff
@@ -40,27 +40,65 @@ test("latex-block renders", async ({ page }) => {
...
+
   // Click text=Home
   await Promise.all([
     page.waitForNavigation(/*{ url: 'http://project-331.local/' }*/),
     page.click("text=Home"),
   ])
   // Click text=University of Helsinki, Department of Mathematics and Statistics
-  await page.click("text=University of Helsinki, Department of Mathematics and Statistics")
+  await Promise.all([
+    page.waitForNavigation,
```
Comment (nygrenh) [inline system-tests/src/tests/latex.spec.ts:99]:
```suggestion
    page.waitForNavigation(),
```

File: system-tests/src/tests/latex.spec.ts [hunk 10]
```diff
@@ -70,8 +108,10 @@ test("latex-block renders", async ({ page }) => {
   // Click button:has-text("Continue")
   await page.click('button:has-text("Continue")')
   // Click text=Chapter 1: first page
-  await page.click("text=Chapter 1: first page")
+  await Promise.all([page.waitForNavigation, page.click("text=Chapter 1: first page")])
```
Comment (nygrenh) [inline system-tests/src/tests/latex.spec.ts:111]:
```suggestion
  await Promise.all([page.waitForNavigation(), page.click("text=Chapter 1: first page")])
```

**Review by nygrenh — CHANGES_REQUESTED** [review]

**Review by nygrenh — CHANGES_REQUESTED** [review]
```js
await runOnlyInHeadlessMode(() => {
const screenshot = x
  expect(page).toMatchSnapshot(...)
})
```

If someone runs the function in headful mode, we will console.warn("Skipping function because we're in headful mode")

File: system-tests/playwright.config.ts [hunk 11]
```diff
@@ -11,7 +11,9 @@ function envToNumber(env: string, defaultNumber: number) {
 const config: PlaywrightTestConfig = {
   globalSetup: require.resolve("./src/setup/globalSetup.ts"),
   globalTeardown: require.resolve("./src/setup/globalTeardown.ts"),
-  use: {},
+  use: {
+    headless: true,
```
Comment (nygrenh) [inline system-tests/playwright.config.ts:15]: make sure this is false when using the --headed flag

**Review by nygrenh — CHANGES_REQUESTED** [review]

File: system-tests/playwright.config.ts (hunk 11 above)
Comment (d471061c) [reply system-tests/playwright.config.ts:15]: it works because headed flag will override this

**Pull request merged**